src/game.h

```diff
@@ -9,6 +9,7 @@
 #include "int_id.h"
 #include "item_location.h"
 #include "cursesdef.h"
+#include "ranged.h"
```

Comment:
Much in agreement, but currently `game::pl_target_ui` and `targeting_data` both use the callback aliases defined in ranged.h.

I think a longer term solution involves reducing the number of files that need to include the header, by deflating `game.h` and creating new classes to take over some of `game`'s functionality.

It shouldn't be too hard to move all the routines involved with player firing/targeting to a new class defined in the new `ranged.h`, then we can reduce `game.h` and `player.h` by a decent amount.